tensorflow-framework/src/main/java/org/tensorflow/framework/activations/ReLU.java

```diff
@@ -0,0 +1,165 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+=======================================================================*/
+package org.tensorflow.framework.activations;
+
+import org.tensorflow.DataType;
+import org.tensorflow.Operand;
+import org.tensorflow.op.Ops;
+import org.tensorflow.op.math.Greater;
+import org.tensorflow.op.nn.LeakyRelu;
+import org.tensorflow.types.family.TNumber;
+
+/**
+ * Rectified Linear Unit(ReLU) activation.
+ *
+ * <p>With default values, this returns the standard ReLU activation: <code>max(x, 0)</code>, the
+ * element-wise maximum of 0 and the input tensor.
+ *
+ * <p>Modifying default parameters allows you to use non-zero thresholds, change the max value of
+ * the activation, and to use a non-zero multiple of the input for values below the threshold.
+ *
+ * <p>For example:
+ *
+ * <pre>
+ *     Operand&lt;TFloat32&gt; input = tf.constant(
+ *              new float[] {-10f, -5f, 0.0f, 5f, 10f});
+ *
+ *     // With default parameters
+ *     ReLU&lt;TFloat32&gt; relu = new ReLU&lt;&gt;(tf);
+ *     Operand&lt;TFloat32&gt; result = relu.call(input);
+ *     // result is [0.f,  0.f,  0.f,  5.f, 10.f]
+ *
+ *     // With alpha = 0.5
+ *     relu = new ReLU&lt;&gt;(tf, 0.5f, ReLU.MAX_VALUE_DEFAULT, ReLU.THRESHOLD_DEFAULT);
+ *     result = relu.call(input);
+ *     // result is [-5.f , -2.5f,  0.f ,  5.f , 10.f]
+ *
+ *     // With maxValue = 5
+ *     relu = new ReLU&lt;&gt;(tf, ReLU.ALPHA_DEFAULT, 5f, ReLU.THRESHOLD_DEFAULT);
+ *     result = relu.call(input);
+ *     // result is [0.f, 0.f, 0.f, 5.f, 5.f]
+ *
+ *     // With threshold = 5
+ *     relu = new ReLU&lt;&gt;(tf, ReLU.ALPHA_DEFAULT, ReLU.MAX_VALUE_DEFAULT, 5f);
+ *     result = relu.call(input);
+ *     // result is [-0.f, -0.f,  0.f,  0.f, 10.f]
+ * </pre>
+ *
+ * @param <T> the data type of the result
+ */
+public class ReLU<T extends TNumber> extends Activation<T> {
+
+  private static final float ALPHA_DEFAULT = 0.0f;
+  private static final Float MAX_VALUE_DEFAULT = null;
+  private static final float THRESHOLD_DEFAULT = 0.0f;
+
+  private final float alpha;
+  private final Float maxValue;
```

Comment:
Can we use `Double.NaN` as a sentinel here rather than a boxed value? Or a boolean?